app/services/post_alerter.rb

```diff
@@ -84,7 +84,8 @@ def self.push_notification(user, payload)
     end
 
     if user.push_subscriptions.exists?
-      if user.seen_since?(SiteSetting.push_notification_time_window_mins.minutes.ago)
+      push_window = SiteSetting.push_notification_time_window_mins
+      if push_window > 0 && user.seen_since?(push_window.minutes.ago)
```

Comment:
When push notification window is 0, we should just run the send push notification job immediately. No need to check when the user was last seen.